### PR TITLE
Make sure annotation data is a valid object

### PR DESCRIPTION
**Pull request by kmshin1397**

### CODE REVIEW

**Review by acrellin — APPROVED** [review]

File: skyportal/handlers/api/annotation.py [hunk 1]
```diff
@@ -115,6 +114,12 @@ def post(self):
             return self.error("Input `origin` must begin with alphanumeric/underscore")
 
         annotation_data = data.get("data")
+
+        if type(annotation_data) is not dict:
```
Comment (dmitryduev) [inline skyportal/handlers/api/annotation.py:118]: Perhaps better do:
```
from typing import Mapping
if not isinstance(annotation_data, Mapping):
...
```
?

Comment (acrellin) [inline skyportal/handlers/api/annotation.py:118]:
```suggestion
        if isinstance(annotation_data, dict):
```
is more idiomatic, but this is fine too

Comment (acrellin) [reply skyportal/handlers/api/annotation.py]: What other data types besides `dict` instances might we expect here?

Comment (dmitryduev) [reply skyportal/handlers/api/annotation.py]: Theoretically something like OrderedDict, but likely we will only see dicts. I just really like this generalization - it's especially nice for Sequences.

**Comment by acrellin** [comment]
@dmitryduev good to merge?

**Pull request merged**